zerver/tests/test_message_fetch.py

```diff
@@ -1534,6 +1545,31 @@ def test_unauthenticated_get_messages_with_web_public(self) -> None:
             result, "Not logged in: API authentication or user session required", status_code=401
         )
 
+    def test_unauthenticated_get_messages_disabled_spectator_login(self) -> None:
```

Comment:
The block of tests around here would benefit from some refactoring.  For a lot of these tests, it's not clear to me that the test would succeed if the thing that we're claiming in the test description is why it's failing is actually the cause of the error, or if the test setup is broken in some other way.

The right way to write such a test suite is to design how the test is written so that each thing that fails is by the construction of the test only in one place different from a similar request that we verify succeeds.

See https://zulip.readthedocs.io/en/latest/testing/philosophy.html#share-test-setup-code.

I think this situation s OK for merging this PR, but we should consider such a refactoring to be a blocker for enabling this in production.

Reply:
Opened this as https://github.com/zulip/zulip/issues/20288.